packages/@ember/canary-features/default-features.d.ts

```diff
@@ -0,0 +1,2 @@
+declare let DEFAULT_FEATURES: any;
```

Comment:
A permissive declaration. What would be the appropriate type definition here? A namespace or interface?

Reply:
Ya, I think you can do:

```js
// packages/@ember/canary-features/default-features.d.ts

export default {
  [key: string]: null | boolean;
}
```

Reply:
Entering that snippet exactly throws some more typescript errors:
```
 yarn lint:tsc
yarn run v1.17.3
$ tsc --noEmit
packages/@ember/canary-features/default-features.d.ts:2:7 - error TS1005: ']' expected.

2   [key: string]:  null | boolean;
        ~

packages/@ember/canary-features/default-features.d.ts:2:15 - error TS1005: ',' expected.

2   [key: string]:  null | boolean;
                ~

packages/@ember/canary-features/default-features.d.ts:2:16 - error TS1136: Property assignment expected.

2   [key: string]:  null | boolean;
                 ~

packages/@ember/canary-features/default-features.d.ts:2:24 - error TS1005: ':' expected.

2   [key: string]:  null | boolean;
                         ~

packages/@ember/canary-features/default-features.d.ts:2:33 - error TS1005: ',' expected.

2   [key: string]:  null | boolean;
```
I'll keep muddling with the syntax in the meantime. :)

Reply:
Whoops, have to use `interface` keyword here:

```ts
export default interface DEFAULT_FEATURES {
  [key: string]: null | boolean;
}
```

Reply:
I think I found the correct combination. I needed to declare an interface and then export a variable that implemented it.